src/electrum.rs

```diff
@@ -266,14 +285,32 @@ impl Rpc {
     ) -> Result<Value> {
         let status = self.new_status(scripthash)?;
         let statushash = status.statushash();
-        client.status.insert(scripthash, status); // skip if already exists
+        client.scripthashes.insert(scripthash, status); // skip if already exists
         Ok(json!(statushash))
     }
 
-    fn new_status(&self, scripthash: ScriptHash) -> Result<Status> {
-        let mut status = Status::new(scripthash);
+    fn outpoint_subscribe(&self, client: &mut Client, (txid, vout): (Txid, u32)) -> Result<Value> {
+        let outpoint = OutPoint::new(txid, vout);
+        let mut status = OutPointStatus::new(outpoint);
+        self.tracker
+            .update_outpoint_status(&mut status, &self.daemon)?;
+        let result = json!(status);
+        client.outpoints.insert(outpoint, status);
```

Comment:
It could be interesting to not subscribe for notifications when the outpoint is spend and the spending transaction has many confirmations (at least 100). This was proposed [here](https://github.com/spesmilo/electrumx/pull/90#discussion_r680886898).

Reply:
Good point - added the following tests to skip index lookups for confirmed funding/spending transactions:
 - https://github.com/romanz/electrs/pull/454/commits/5d2c7cfd26be133fdc0e9701626f3e0970904798#diff-0bfaab7c77d86eb41c2fe32c5e5df509fa02ab04c97b0ce3216911afaae0d67cR527
 - https://github.com/romanz/electrs/pull/454/commits/5d2c7cfd26be133fdc0e9701626f3e0970904798#diff-0bfaab7c77d86eb41c2fe32c5e5df509fa02ab04c97b0ce3216911afaae0d67cR562

Reply:
Ok so if I understand the code well, it will still subscribe for notifications even if the outpoint is spent with many confirmations. However OutpointStatus syncing will only look at the index once, then it only returns its previous value, except if a reorg is detected. If adding notification has no cost if they don't require index lookup, that's nice :)